### PR TITLE
Updated design docs structure and added new content

### PR DESCRIPTION
**Pull request by heatherlarsen**
## :pushpin: Summary

If merged, this PR would update the design documentation on the microsite to include more robust documentation from @dizzyup (previously added to Figma), updated hierarchy of content, and improved formatting. There are a few known improvements that need to be made (see Detailed Description).

## :hammer_and_wrench: Detailed Description

- [x] Added new content from Figma
- [x] Improved hierarchy for better information flow
- [x] Cleaned up some of the formatting (mostly around the lists)
- [x] Fix minor accessibility issues (skipped headings, icon/image alts, etc)
- [x] Check responsiveness

#### Reprioritized for after 1.0
- [x] ~~Fix image sizing & alignment issues~~
- [x] ~~Add section for immediate icon usage guidelines (ie. don't have time to submit to DS Team)~~

## :camera_flash: Screenshots

Screen Shot 2021-09-17 at 4 09 55 PM

### :speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

Examples: 
- issue (ux,non-blocking): These buttons should be red, but let's handle this in a follow-up.
- suggestion (non-blocking): Let's change this wording to make it easier to understand.
- issue (blocking): We shouldn't introduce this kind of tech debt; let's pair and resolve the issue in a more sustainable way.

### CODE REVIEW

**Review by didoo — COMMENTED** [review]
In general is OK, definitely more clear now. One thing we may be missing is the description of what to do if a designer needs to use the icon immediately in their designs (as instance of a published component). This was a real, very common use case in my previous company. Maybe here is not. Not sure.

**Review by didoo — APPROVED** [review]

**Comment by heatherlarsen** [comment]
@Dhaulagiri @MelSumner @dizzyup any feedback/changes needed here?

File: ember-flight-icons/tests/dummy/app/templates/design.hbs [hunk 1]
```diff
@@ -1,441 +1,683 @@
...
-            alt=""
-            role="none"
-          />
+            <img class="w-28" src="assets/images/icon-types-st.png" alt="" role="none" />
+        </div>
+      </div>
+    </div>
+    <div class="text-left sm:flex lg:block">
+      <div class="mt-3 sm:flex-auto sm:mt-0 sm:mr-6 lg:mt-6 lg:mr-0">
+        <h4 class="text-xl font-medium text-gray-900">Contained</h4>
```
Comment (MelSumner) [inline ember-flight-icons/tests/dummy/app/templates/design.hbs:66]: question: Can we use the classes we've already defined?
```suggestion
        <h4 class="ds-h4">Contained</h4>
```

File: ember-flight-icons/tests/dummy/app/templates/design.hbs [hunk 2]
```diff
@@ -1,441 +1,683 @@
...
-      <div class="mt-3 sm:mt-0 sm:ml-6 lg:mt-6 lg:ml-0">
-        <h4 class="text-lg font-medium text-gray-900">
-          Off
-        </h4>
-        <p class="mt-2 text-sm text-gray-500">
-          Glyph with a strike-through; used to indicate a disabled state.
+    <div class="text-left sm:flex lg:block">
+      <div class="mt-3 sm:flex-auto sm:mt-0 sm:mr-6 lg:mt-6 lg:mr-0">
+        <h4 class="text-xl font-medium text-gray-900">Off</h4>
+        <p class="text-sm text-gray-500">
```
[the remark this answers is not in the document]
Comment (heatherlarsen) [reply ember-flight-icons/tests/dummy/app/templates/design.hbs:51]: contrast checkers all say it passes. can you please elaborate?

**Comment by amyrlam** [comment]
i added prettier to hbs here https://github.com/hashicorp/flight/pull/233 

if you have any issues with linting after the rebase, you should be able to run `yarn run lint:hbs:fix` on terminal (from `cd ember-flight-icons/`) per https://github.com/hashicorp/flight/blob/main/ember-flight-icons/package.json#L48. let me know if you need any help with that stuff once this is ready to be merged in

FWIW i think if you open a PR as draft first, CI doesn't run re: ^^ "waiting for status to be reported". it's very confusing

File: ember-flight-icons/tests/dummy/app/templates/design.hbs (hunk 2 above)
[the remark this answers is not in the document]
Comment (MelSumner) [reply ember-flight-icons/tests/dummy/app/templates/design.hbs:51]: Sure, color contrast checker tests are run on assumed font sizes. Generally, the smaller a font gets, the darker it should be.

File: ember-flight-icons/tests/dummy/app/templates/design.hbs (hunk 1 above)
Comment (MelSumner) [reply ember-flight-icons/tests/dummy/app/templates/design.hbs]: ok lets adjust that one in a separate PR. 👍

File: ember-flight-icons/tests/dummy/app/templates/design.hbs (hunk 2 above)
[the remark this answers is not in the document]
Comment (heatherlarsen) [reply ember-flight-icons/tests/dummy/app/templates/design.hbs:53]: will get it updated this afternoon. i'd love to learn more about this. can you please share some resources?

**Review by MelSumner — APPROVED** [review]
Let's ship it! We can iterate. 👍

File: ember-flight-icons/tests/dummy/app/templates/design.hbs (hunk 1 above)
Comment (heatherlarsen) [reply ember-flight-icons/tests/dummy/app/templates/design.hbs]: I already took care of this one, just need to fix the linting issues.

**Pull request merged**